app/src/main/java/org/oppia/android/app/hintsandsolution/HintsViewModel.kt

```diff
@@ -28,7 +30,10 @@ private const val DEFAULT_HINT_AND_SOLUTION_SUMMARY = ""
 @FragmentScope
 class HintsViewModel @Inject constructor(
   private val resourceHandler: AppLanguageResourceHandler,
-  private val translationController: TranslationController
+  private val translationController: TranslationController,
+  private val htmlParserFactory: HtmlParser.Factory,
+  @DefaultResourceBucketName private val resourceBucketName: String,
+  @ExplorationHtmlParserEntityType private val entityType: String
```

Comment:
I think this changes based on whether an exploration or question is being loaded. We probably need to pass in the entity type from the fragment (which I expect we must do for the content itself since hints can contain images).

Reply:
I am not able to understand that how can I use entity type from fragment. Can you elaborate a little bit. Thanks @BenHenning